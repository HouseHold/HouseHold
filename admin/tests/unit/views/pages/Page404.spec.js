import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Page404 from '@/views/pages/Page404'

Vue.use(CoreuiVue)

describe('Page404.vue', () => {
  it('has a name', () => {
    expect(Page404.name).toBe('Page404')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Page500', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.is(Page404)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.find('h1').text()).toMatch('404')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.element).toMatchSnapshot()
  })
})
