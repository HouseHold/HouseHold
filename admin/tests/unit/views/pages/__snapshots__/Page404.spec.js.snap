// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404.vue renders correctly 1`] = `
<ccontainer-stub
  class="d-flex align-items-center min-vh-100"
  tag="div"
>
  <crow-stub
    class="w-100 justify-content-center"
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="6"
      tag="div"
    >
      <div
        class="w-100"
      >
        <div
          class="clearfix"
        >
          <h1
            class="float-left display-3 mr-4"
          >
            404
          </h1>
           
          <h4
            class="pt-3"
          >
            Oops! You're lost.
          </h4>
           
          <p
            class="text-muted"
          >
            The page you are looking for was not found.
          </p>
        </div>
         
        <cinput-stub
          class="mb-3"
          lazy="400"
          placeholder="What are you looking for?"
          type="text"
        />
      </div>
    </ccol-stub>
  </crow-stub>
</ccontainer-stub>
`;
